kentikapi/internal/synthetics/test_results.go

```diff
@@ -0,0 +1,287 @@
+	if err != nil {
+		return nil, err
+	}
+
+	jitter, err := metricDataFromPayload(r.Ping.Jitter)
+	if err != nil {
+		return nil, err
+	}
+
+	dstIP := net.ParseIP(r.Ping.DstIp)
+	if dstIP == nil {
+		return nil, fmt.Errorf("cannot parse ping destination IP %q", r.Ping.DstIp)
+	}
+
+	return synthetics.PingResults{
+		Target:     r.Ping.Target,
+		PacketLoss: packetLoss,
+		Latency:    latency,
+		Jitter:     jitter,
+		DstIP:      dstIP,
+	}, nil
+}
+
+func httpTaskResultsFromPayload(r *syntheticspb.TaskResults_Http) (synthetics.TaskSpecificResults, error) {
+	if r.Http == nil {
+		return nil, fmt.Errorf("HTTP task specific results are nil")
+	}
+
+	target, err := url.Parse(r.Http.Target)
+	if err != nil {
+		return nil, err
+	}
+
+	latency, err := metricDataFromPayload(r.Http.Latency)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := httpResponseDataFromPayload(r.Http.Response)
+	if err != nil {
+		return nil, err
+	}
+
+	dstIP := net.ParseIP(r.Http.DstIp)
+	if dstIP == nil {
+		return nil, fmt.Errorf("cannot parse HTTP destination IP %q", r.Http.DstIp)
+	}
+
+	return synthetics.HTTPResults{
+		Target:   *target,
+		Latency:  latency,
+		Response: response,
+		DstIP:    dstIP,
+	}, nil
+}
+
+func dnsTaskResultsFromPayload(r *syntheticspb.TaskResults_Dns) (synthetics.TaskSpecificResults, error) {
+	if r.Dns == nil {
+		return nil, fmt.Errorf("DNS task specific results are nil")
+	}
+
+	server := net.ParseIP(r.Dns.Server)
+	if server == nil {
+		return nil, fmt.Errorf("cannot parse DNS destination IP %q", r.Dns.Server)
+	}
+
+	latency, err := metricDataFromPayload(r.Dns.Latency)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := dnsResponseDataFromPayload(r.Dns.Response)
+	if err != nil {
+		return nil, err
+	}
+
+	return synthetics.DNSResults{
+		Target:   r.Dns.Target,
+		Server:   server,
+		Latency:  latency,
+		Response: response,
+	}, nil
+}
+
+func packetLossDataFromPayload(d *syntheticspb.PacketLossData) (synthetics.PacketLossData, error) {
+	if d == nil {
+		return synthetics.PacketLossData{}, fmt.Errorf("packet loss data is nil")
+	}
+
+	return synthetics.PacketLossData{
+		Current: d.Current,
+		Health:  synthetics.Health(d.Health),
+	}, nil
+}
+
+func metricDataFromPayload(d *syntheticspb.MetricData) (synthetics.MetricData, error) {
+	if d == nil {
+		return synthetics.MetricData{}, fmt.Errorf("metric data is nil")
+	}
+
+	return synthetics.MetricData{
+		Current:       time.Duration(d.Current) * time.Microsecond,
+		RollingAvg:    time.Duration(d.RollingAvg) * time.Microsecond,
+		RollingStdDev: time.Duration(d.RollingStddev) * time.Microsecond,
+		Health:        synthetics.Health(d.Health),
+	}, nil
+}
+
+func httpResponseDataFromPayload(d *syntheticspb.HTTPResponseData) (synthetics.HTTPResponseData, error) {
+	if d == nil {
+		return synthetics.HTTPResponseData{}, fmt.Errorf("HTTP response data is nil")
+	}
+
+	return synthetics.HTTPResponseData{
+		Status: d.Status,
+		Size:   d.Size,
+		Data:   d.Data,
```

Comment:
The `data` attribute (if non-empty) actually contains JSON payload, which we should parse. Example:
```
  - id: '56114'
    health: critical
    tasks:
    - type: http
      target: http://e1-rp-nyc1.edgeuno.net [149.112.2.35]
      latency: 'current: 282.35ms avg: 281.62ms stddev: 0.723ms (healthy)'
      response:
        status: 404
        size: 153
        data:
          duration: 282346.5
          request_start: 142343
          https_validity: 0
          https_expiry_timestamp: 0
          errors: []
          connect_end: 142343
          domain_lookup_end: 2376.5
```

Reply:
Done: 8ca507275e7c9dc21cbf57b69ff80d7169b58b2e

I modified the code to un-marshall data and convert map keys to snake case. Is it good enough?